### PR TITLE
[Spring] add --deployment-name for app create

### PR DESCRIPTION
**Pull request by RuoyuWang-MS**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az spring app create` command add --deployment-name argument. 

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring

File: src/spring/azext_spring/_params.py [hunk 1]
```diff
@@ -268,6 +268,8 @@ def load_arguments(self, _):
                    help='A json file path for the persistent storages to be mounted to the app')
         c.argument('loaded_public_certificate_file', options_list=['--loaded-public-certificate-file', '-f'], type=str,
                    help='A json file path indicates the certificates which would be loaded to app')
+        c.argument('deployment_name', type=str,
+                   help='Name of the active deployment. Deployment named "default" will be created if this argument not provided', validator=validate_name)
```
Comment (allxiao) [inline src/spring/azext_spring/_params.py:272]: We should use the named argument `default="default"` to set the default value. With this we don't need to highlight it in the help text.

Check the `--sku` argument for reference. In the help output, it shows:

```
    --sku                                                   : Name of SKU. Enterprise is still in
                                                              Preview.  Allowed values: Basic,
                                                              Enterprise, Standard.  Default:
                                                              Standard.
```

**Review by allxiao — CHANGES_REQUESTED** [review]
please check the argument spec

File: src/spring/azext_spring/_params.py (hunk 1 above)
Comment (RuoyuWang-MS) [reply src/spring/azext_spring/_params.py]: Good point! I will modify the help text as follows:
```
    --deployment-name                                           : Name of the active deployment.
                                                                  Default: default.
```

File: src/spring/azext_spring/_params.py [hunk 2]
```diff
@@ -268,6 +268,8 @@ def load_arguments(self, _):
                    help='A json file path for the persistent storages to be mounted to the app')
         c.argument('loaded_public_certificate_file', options_list=['--loaded-public-certificate-file', '-f'], type=str,
                    help='A json file path indicates the certificates which would be loaded to app')
+        c.argument('deployment_name', type=str, default='default',
+                   help='Name of the active deployment.', validator=validate_name)
```
Comment (allxiao) [inline src/spring/azext_spring/_params.py:272]: is this `active deployment` or `default deployment`? According to line 148 in app.py, this should be `default deployment`.

Comment (RuoyuWang-MS) [reply src/spring/azext_spring/_params.py]: Sure. Will change to `default deployment`

**Review by allxiao — APPROVED** [review]

File: src/spring/azext_spring/_params.py [hunk 3]
```diff
@@ -267,6 +267,8 @@ def load_arguments(self, _):
                    help='A json file path for the persistent storages to be mounted to the app')
         c.argument('loaded_public_certificate_file', options_list=['--loaded-public-certificate-file', '-f'], type=str,
                    help='A json file path indicates the certificates which would be loaded to app')
+        c.argument('deployment_name', type=str, default='default',
+                   help='Name of the default deployment.', validator=validate_name)
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_params.py:271]: `str` is the default type, so it does not need to be defined repeatedly
```suggestion
        c.argument('deployment_name', default='default',
                   help='Name of the default deployment.', validator=validate_name)
```

File: src/spring/HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 Release History
 ===============
+1.7.0
+---
+* Add argument `--deployment-name` in command `az spring app create`.
```
Comment (zhoxing-ms) [inline src/spring/HISTORY.md:5]: Do we need to upgrade the secondary version just for adding a new parameter?

File: src/spring/azext_spring/_params.py (hunk 3 above)
Comment (RuoyuWang-MS) [reply src/spring/azext_spring/_params.py]: Thanks for your suggestion. Will do

File: src/spring/HISTORY.md (hunk 4 above)
Comment (RuoyuWang-MS) [reply src/spring/HISTORY.md]: Sure. Will change 1.7.0 to 1.6.4.

**Comment by RuoyuWang-MS** [comment]
@zhoxing-ms Hi Xing. Would you please help review it again when you are free? Please merge this PR if it is ready to merge. Thanks a lot!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**